source/engine/graphics/system/ParticleEmitterSystem.cpp

```diff
@@ -1,129 +1,210 @@
 #include "ParticleEmitterSystem.h"
+#include "assets/AssetService.h"
```

Comment:
Sorry, my unindenting kind of hides relevant changes. Other than the new constructor member, changes should be in `UpdateParticles()`, `SortEmitters()`, and `Execute()`.